packages/mp/src/mp/validate/pre_build_validation/version_bump_validation.py

```diff
@@ -0,0 +1,188 @@
+            raise NonFatalValidationError(msg)
+
+        existing_files, new_files = _create_data_for_version_bump_validation(rn_path, toml_path)
+        _version_bump_validation_run_checks(existing_files, new_files)
+
+
+def _create_data_for_version_bump_validation(
+    rn_path: pathlib.Path, toml_path: pathlib.Path
+) -> VersionBumpValidationData:
+    existing_files: ExistingIntegrationFiles = {
+        "toml": TomlFileVersions(),
+        "rn": ReleaseNoteFileVersions(),
+    }
+    new_files: NewIntegrationFiles = NewIntegrationFiles()
+
+    try:
+        old_toml_content = mp.core.unix.get_file_content_from_main_branch(toml_path)
+        existing_files["toml"]["old"] = PyProjectToml.from_toml_str(old_toml_content)
+        existing_files["toml"]["new"] = PyProjectToml.from_toml_str(
+            toml_path.read_text(encoding="utf-8")
+        )
+
+        old_rn_content = mp.core.unix.get_file_content_from_main_branch(rn_path)
+        existing_files["rn"]["old"] = _get_last_note(old_rn_content)
+        existing_files["rn"]["new"] = _get_new_rn_notes(
+            rn_path.read_text(encoding="utf-8"), old_rn_content
+        )
+
+    except mp.core.unix.NonFatalCommandError:
+        new_files["toml"] = PyProjectToml.from_toml_str(toml_path.read_text(encoding="utf-8"))
+        new_files["rn"] = ReleaseNote.from_non_built_str(rn_path.read_text(encoding="utf-8"))
+
+    return existing_files, new_files
+
+
+def _get_last_note(content: str) -> ReleaseNote | None:
+    notes = ReleaseNote.from_non_built_str(content)
+    return notes[-1] if notes else None
+
+
+def _get_new_rn_notes(new_rn_content: str, old_rn_content: str) -> list[ReleaseNote]:
+    new_notes: list[ReleaseNote] = ReleaseNote.from_non_built_str(new_rn_content)
+    old_notes: list[ReleaseNote] = ReleaseNote.from_non_built_str(old_rn_content)
+    return new_notes[len(old_notes) :]
+
+
+def _version_bump_validation_run_checks(
+    existing_files: ExistingIntegrationFiles, new_files: NewIntegrationFiles
+) -> None:
+    msg: str
+    new_notes: list[ReleaseNote]
+    if (new_toml := existing_files["toml"].get("new")) and (
+        old_toml := existing_files["toml"].get("old")
+    ):
+        toml_new_version = new_toml.project.version
+        toml_old_version = old_toml.project.version
+
+        if toml_new_version != toml_old_version + 1.0:
+            msg = "The project.toml Version must be incremented by exactly 1.0."
+            raise NonFatalValidationError(msg)
+
+        new_notes = existing_files["rn"].get("new")
+        msg = (
+            "The release note's version must match the new version of the project.toml and be "
+            "consistent in all the newly added notes."
+        )
+        if not _rn_is_valid(new_notes, toml_new_version):
+            raise NonFatalValidationError(msg)
+
+    elif new_files.get("toml") and new_files.get("rn"):
+        toml_version = new_files["toml"].project.version
+        new_notes = new_files["rn"]
```

Comment:
Can this be shortened?

```suggestion
    elif new_toml := new_files.get("toml") and (new_notes := new_files.get("rn")):
        toml_version = new_toml.project.version
```